infra/gcp/modules/gke-cluster/main.tf

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Create GCP SA for nodes
+resource "google_service_account" "cluster_node_sa" {
+  project      = var.project_name
+  account_id   = "gke-nodes-${var.cluster_name}"
+  display_name = "Nodes in GKE cluster '${var.cluster_name}'"
+}
+
+// Add roles for SA
+resource "google_project_iam_member" "cluster_node_sa_logging" {
+  project = var.project_name
+  role    = "roles/logging.logWriter"
```

Comment:
You should consider making logWriter and metricWriter standard on all your SA's. It enables logging for them and adds the benefit of making the ServiceAccount's visible GUI in the IAM console under IAM. The visibility under IAM makes it easier to visibly audit what roles a SA has. The console hides role mappings in the service account screen.